src/sdltiles.cpp

```diff
@@ -873,6 +883,13 @@ bool Font::draw_window( WINDOW *win, int offsetx, int offsety )
     //Specifically when showing the overmap
     //And in some instances of screen change, i.e. inventory.
     bool oldWinCompatible = false;
+
+    // use the oversize buffer when dealing with windows that can have a different font than the main text font
+    std::vector<curseline> *bufferpointer = &terminal_framebuffer;
+    if( g && ( win == g->w_terrain || win == g->w_overmap ) ) {
+        bufferpointer = &oversized_framebuffer;
+    }
```

Comment:
`bufferpointer` could be a reference, initialized via a ternary:
```C++
const bool use_terminal_buffer = ...;
auto &framebuffer = use_terminal_buffer ? terminal_framebuffer : oversized_framebuffer;
```
And you don't even have to change the line below where it's used.